botocore/endpoint.py

```diff
@@ -152,16 +130,13 @@ def _create_request_object(self, request_dict):
                              headers=headers)
         return request
 
-    def prepare_request(self, request, signer):
-        if signer is not None:
-            with self._lock:
-                # Parts of the auth signing code aren't thread safe (things
-                # that manipulate .auth_path), so we're using a lock here to
-                # prevent race conditions.
-                event_name = 'before-auth.%s' % self._endpoint_prefix
-                self._event_emitter.emit(
-                    event_name, endpoint=self, request=request, auth=signer)
-                signer.add_auth(request=request)
+    def prepare_request(self, request, operation_model=None):
+        if operation_model:
+            event_name = 'request-created.{endpoint_prefix}.{op_name}'.format(
```

Comment:
It is kind of weird that the event is called ``requested-created`` and it is located in the ``prepare_request`` method. Maybe make it ``request-prepared``? If you really need the ``request-created`` event maybe put it in the ``create_request`` method for the sake of making the code easier to understand.